backend/users-service/app/src/index.ts

```diff
@@ -1,7 +1,71 @@
-import { Elysia } from "elysia";
+import { Elysia, t } from 'elysia'
+import { swagger } from '@elysiajs/swagger'
+import { cors } from '@elysiajs/cors'
+import { helmet } from 'elysia-helmet'
+import jwt from '@elysiajs/jwt'
+import { PrismaClient } from '@prisma/client'
+import { logger } from '@bogeychan/elysia-logger'
+import { Role } from './constants'
 
-const app = new Elysia().get("/", () => "Users service").listen(80);
+const db = new PrismaClient()
+
+const app = new Elysia({ prefix: '/users' })
+  .use(cors())
+  .use(helmet())
+  .use(
+    jwt({
+      name: 'jwt',
+      secret: process.env.JWT_SECRET as string
+    })
+  )
+  .use(logger())
+  .use(swagger())
+  .get(
+    '/:id',
+    async ({ params: { id } }) =>
+      await db.user.findUnique({
+        where: {
+          id
+        },
+        include: {
+          questions: true
+        }
+      })
+  )
+  .post(
+    '/',
+    async ({ body: { avatarUrl, email } }) =>
+      await db.user.create({
+        data: {
+          avatarUrl,
+          email,
+          role: Role.User
+        }
+      }),
+    {
+      body: t.Object({ avatarUrl: t.String(), email: t.String() })
+    }
+  )
+  .put(
```

Comment:
this `put` request doesn't seem complete and invokes the Prisma `create` instead of `update` - may I know what you had in mind for this endpoint?

Reply:
> this `put` request doesn't seem complete and invokes the Prisma `create` instead of `update` - may I know what you had in mind for this endpoint?

Incomplete. Resolved with fe3dbd3d5f5bc4bf4b72dccd4f034f6920e3481f